src/ol/render/canvas/Executor.js

```diff
@@ -75,6 +75,20 @@ function getDeclutterBox(replayImageOrLabelArgs) {
   return replayImageOrLabelArgs[3].declutterBox;
 }
 
+const rtlRegEx = /[\u0591-\u07FF]/;
```

Comment:
```suggestion
const rtlRegEx = new RegExp(
  /* eslint-disable prettier/prettier */
  '[' +
    String.fromCharCode(0x00591) + '-' + String.fromCharCode(0x008ff) +
    String.fromCharCode(0x0fb1d) + '-' + String.fromCharCode(0x0fdff) +
    String.fromCharCode(0x0fe70) + '-' + String.fromCharCode(0x0fefc) +
    String.fromCharCode(0x10800) + '-' + String.fromCharCode(0x10fff) +
    String.fromCharCode(0x1e800) + '-' + String.fromCharCode(0x1efff) +
  ']'
  /* eslint-enable prettier/prettier */
);
```
Apparently it is  not possible to use 5-digit unicode escape codes in js ...

The data I used to look it up is this:
https://github.com/openlayers/openlayers/blob/e7a23d619adc41f8ebe7b57ac5e42fe528533c1c/unicode14.ods?raw=true
sourced from: https://www.unicode.org/Public/14.0.0/ucd/UnicodeData-14.0.0d3.txt

RTL characters are of type `AL` or `R` (orange in the spreadsheet),  LTR characters are of type `L` (blue in the spreadsheet), see: https://www.unicode.org/reports/tr9/#Table_Bidirectional_Character_Types

Reply:
Thanks for the thorough research on this!